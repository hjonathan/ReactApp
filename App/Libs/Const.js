import RNFS from "react-native-fs";

export const CASE = {
    INBOX: "Inbox",
    DRAFT: "Draft",
    PARTICIPATED: "Participated",
    UNASSIGNED: "Unassigned",
    STATUS: {
        NONE: "none",
        SENDING: "sending",
        WORKING: "working",
        OFFLINE: "offline"
    }
};

export const DATETIME = {
    LOCAL: "YYYY-MM-DD HH:mm:ss",
    LOCAL_UTC: "YYYY-MM-DD HH:mm:ss Z"
};

export const DATE = {
    SIMPLE: "YYYYMMDD",
    LOCAL: "YYYY-MM-DD"
};

export const TIME = {
    LOCAL: "HH:mm:ss"
};

export const DS = "/";

export const PATH = {
    PROJECT: `${RNFS.DocumentDirectoryPath}`,
    FORMRENDER: `${DS}FormRender`,
    STEPS: `${DS}Steps`,
    NEXTSTEPS: `${DS}NextSteps`,
    FROMS: `${DS}Forms`,
    DATA: `${DS}Data`,
    EXTERNALLIBS: `${DS}ExternalLibs`,
    BUILDPROD: `${DS}build-prod`,
    MUSIC: "/storage/emulated/0/Music/processmakeraudiorecord/"
};

export const FILENAMES = {
    EXTERNALMAP: "ExtMap.json"
};
