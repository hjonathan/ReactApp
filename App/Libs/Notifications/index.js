/* eslint-disable import/prefer-default-export */
import * as pushNotification from "./pushNotification";

export {pushNotification};
