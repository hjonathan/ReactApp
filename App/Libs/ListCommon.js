import _ from "lodash";
import Task from "../Model/Task";
import Step from "../Model/Step";
import {CASE} from "./Const";

/**
 * Function commons for: inbox, draft, unassigned and participated
 */
export default class ListCommon {
    /**
     * Constructor
     * @param props
     */
    constructor (props) {
        this.props = props;
        this.task = Task.allTaskIdOffline();
    }

    /**
     * Get data lists
     * @param method
     * @param params
     */
    getDataList (method, params = {}) {
        if (this.props[method]) {
            this.props[method](params);
        }
    }

    /**
     * Gets usersInfo
     * @param caseList
     * @param type is type in list: prevUser, currentUser
     */
    getUsersInfo (caseList, type) {
        if (caseList.length > 0 && caseList[0][type]) {
            const ids = caseList.map(cases => cases[type].userId);
            this.props.userInfoRequest({
                size: "1",
                user: {
                    ids: _.uniq(ids)
                }
            });
        }
    }

    /**
     * Merges userPhoto props to dataList
     * @param params
     * @param type is type in list: prevUser, currentUser
     */
    mergeList (params, type) {
        let i,
            j,
            cbUser = function (user) { return user.userId === params.caseList[i][type].userId; };
        if (params.caseList) {
            for (i = 0; i < params.caseList.length; i += 1) {
                if (params.caseList[i][type]) {
                    j = _.findIndex(params.users, cbUser);
                    if (j >= 0) {
                        params.caseList[i].userPhoto = params.users[j].userPhoto;
                    }
                    params.caseList[i].status = this.processIfOffline(params.caseList[i]);
                }
            }
        }
        return params.caseList;
    }

    /**
     * Check if case is offline and get all necessary for case
     * @param caseData
     * @returns {boolean}
     */
    processIfOffline (caseData) {
        let isOffline,
            {status} = caseData;
        if (!status || status === CASE.STATUS.NONE) {
            if (caseData.task.offlineEnabled === "TRUE" && this.props.caseOffline) {
                this.props.caseOffline(caseData);
                status = CASE.STATUS.OFFLINE;
            }
            isOffline = Step.existSteps(caseData.process.processId, caseData.task.taskId);
            if (isOffline && _.includes(this.task, caseData.task.taskId)) {
                status = CASE.STATUS.OFFLINE;
            }
        }
        return status;
    }

    /**
     * Key for FlatList formed by delIndex and caseId
     * @param item
     * @returns {string}
     */
    keyExtractor (item, index) {
        return `${item.delIndex}_${item.caseId}`;
    }
}
