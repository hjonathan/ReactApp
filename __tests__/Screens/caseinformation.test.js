import React from 'react'
import { shallow } from 'enzyme';
import ConnectedCaseInformation from '../../App/Screens/CaseInformation';
import CaseInformationView from '../../App/Screens/CaseInformation/caseinformation';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
jest.mock('react-native-orientation', () => require.requireActual('../../__mocks__/react-native-orientation').default);

import { createStore } from 'redux'
import {CASE} from "../../App/Libs/Const";

describe('Screen >>> Case Information', () => {
    const initialState = {
            Cases: {
                info: {
                    case: {
                        caseId: "123456678",
                        caseTitle: "Case title for test"
                    },
                    task: {
                        taskId: "1234567890",
                        taskTitle: "Task title for test"
                    }
                }
            },
            promptTitle: '',
            case: 'null',
            task: 'null',
            toastMessage: 'toast test',
            loader: false
    };
    let store, container, viewCase;
    const parseValueCreator = () => { },
    mockStore = configureStore(),
    caseClaimRequest = () => { app_uid: '' },
    caseInfoRequest = () => { },
    navigation = {
        state: {
            params: {
                type: CASE.PARTICIPATED,
                caseId: ''
            }
        },
        dispatch: jest.fn()
    };

    beforeEach(() => {
        store = mockStore(initialState)
        container = shallow(<ConnectedCaseInformation store={store} />)
        viewCase = shallow(<CaseInformationView parseValueCreator={parseValueCreator} navigation={navigation} caseClaimRequest={caseClaimRequest} caseInfoRequest={caseInfoRequest} store={store}/>)
    })

    /**
    * This test is related to the render of the WalkThrough view.
    */
    it('render Case Information correctly', () => {
        expect(toJson(container)).toMatchSnapshot();
    });

    it('Render the connected(SMART) component', () => {
        expect(container.length).toEqual(1)
    });

    it('Check Prop matches with initialState', () => {
        expect(container.prop('info').case).toEqual({
            caseId: "123456678",
            caseTitle: "Case title for test"
        });
    });

    it('Check static navigationOptions', () => {
        expect(CaseInformationView.navigationOptions(""))
            .toEqual({
                title: "general_information",
                headerTintColor: "#FFF",
                headerStyle: {backgroundColor: "#1F6CBC"}
            });
    });

    it('render claim button incorrect param', () => {
        expect(viewCase.instance().renderClaimCaseButton('')).toBeNull();
    });

    it('render test valid param', () => {
        expect(viewCase.instance().renderClaimCaseButton('Unassigned')).not.toBeNull();
    });

    it('Props coverage willReceiveProps', () => {
        viewCase.instance().componentWillReceiveProps({info: {
            case: {
                caseId: "123456678",
                caseTitle: "Case title for test"
            },
            task: {
                taskId: "1234567890",
                taskTitle: "Task title for test"
            }
        }});
        expect(viewCase.instance().state.case.caseId).toEqual("123456678");
    });

    it('Props coverage componentDidUpdate', () => {
        viewCase.instance().componentDidUpdate();
        expect(viewCase.instance().state.toastMessage).toBeNull();
    });

    it('Props coverage claimCaseFromCase', () =>{
        viewCase.instance().state.case = {
            caseId: "123456678",
            caseTitle: "Case title for test",
            delIndex: 1
        };
        viewCase.instance().state.task = { taskId: "123456678"};
        viewCase.instance().claimCaseFromCase();
        expect(viewCase.instance().state.loader).toEqual(true);
    });

    it('validator To do Text', () => {
        expect(container.prop('parseValueCreator')('TO_DO')).toEqual('To do');
    });

    it('validator In Progress Text', () => {
        expect(container.prop('parseValueCreator')('in_progress')).toEqual('In progress');
    });

    it('validator Reasigned Text', () => {
        expect(container.prop('parseValueCreator')('reasigned')).toEqual('Reasigned');
    });

    it('validator Routed Text', () => {
        expect(container.prop('parseValueCreator')('null')).toEqual('Routed');
    });

    it('validator null Text', () => {
        expect(container.prop('parseValueCreator')()).toBeNull();
    });
});
