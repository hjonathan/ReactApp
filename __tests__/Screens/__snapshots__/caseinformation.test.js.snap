// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen >>> Case Information render Case Information correctly 1`] = `
<CaseInformation
  caseClaimRequest={[Function]}
  caseInfoRequest={[Function]}
  error={null}
  info={
    Object {
      "case": Object {
        "caseId": "123456678",
        "caseTitle": "Case title for test",
      },
      "task": Object {
        "taskId": "1234567890",
        "taskTitle": "Task title for test",
      },
    }
  }
  parseValueCreator={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
